homework_1_collections/Nucleo.py

```diff
@@ -0,0 +1,135 @@
+# Transcribes DNA to RNA.
+# dna_seq - string, DNA sequence.
+def transcribe(dna_seq):
+    x = list(dna_seq)
+    for i in range(len(x)):
+        if x[i] == 'T':
+            x[i] = 'U'
+        if x[i] == 't':
+            x[i] = 'u'
+    x = (''.join(x))
+    return x
+
+
+# Reverses the nucleic acid sequence.
+# seq - string, DNA or RNA sequence.
+def reverse(seq):
+    return seq[::-1]
+
+
+# Builds and returns a complementary strand of DNA or RNA.
+# seq - string, DNA or RNA sequence; rna=True for RNA.
+def complement(seq, rna=False):
+    x = list(seq)
+    main_code = {
+        'A': 'T', 'a': 't',
+        'T': 'A', 't': 'a',
+        'G': 'C', 'g': 'c',
+        'C': 'G', 'c': 'g'
+    }
+
+    code_rna = {
+        'A': 'U', 'a': 'u',
+        'U': 'A', 'u': 'a',
+        'G': 'C', 'g': 'c',
+        'C': 'G', 'c': 'g'
+    }
+
+    if rna:
+        main_code = code_rna
+    new_seq = []
+    for i in x:
+        i = main_code[i]
+        new_seq.append(i)
+    x = (''.join(new_seq))
+    return x
+
+
+# Checks nucleic acid sequence.
+# seq - string, DNA or RNA sequence.
+def check_your_seq(seq):
+    # Valid sequence elements.
+    list_of_na = ['A', 'C', 'G', 'T', 'U', 'a', 'c', 'g', 't', 'u']
+    x = True
+    for i in list(seq):
+        if i not in list_of_na:
+            x = False
+    # Uracil and thymine cannot be in the same sequence.
+    for i in list(seq.lower()):
+        if i == 't':
+            for j in list(seq.lower()):
+                if j == 'u':
+                    x = False

```

Comment:
Мы не проходили, но здесь отлично работают множества.
```suggestion
    set_of_na = {'A', 'C', 'G', 'T', 'U', 'a', 'c', 'g', 't', 'u'}
    x = True
    if not set(seq) <= set_of_na:
        x = False
    if set(seq.lower()) >= {'u', 't'}:
        x = False
```
\>= и <= - операторы проверки на подмножества